test/attr/backDeploy.swift

```diff
@@ -0,0 +1,130 @@
+// RUN: %target-typecheck-verify-swift -parse-as-library
+
+// MARK: - Valid declarations
+
+@available(macOS 12.0, *)
+@_backDeploy(macOS 11.0)
+public func backDeployedTopLevelFunc() {}
+
+// FIXME(backDeploy): Availability macros should be supported
+
+public class TopLevelClass {
+  @available(macOS 12.0, *)
+  @_backDeploy(macOS 11.0)
+  public func backDeployedMemberFunc() {}
+
+  // FIXME(backDeploy): Computed properties should be supported
+  @available(macOS 12.0, *)
+  @_backDeploy(macOS 11.0) // expected-error {{'@_backDeploy' attribute cannot be applied to this declaration}}
```

Comment:
It would be nice to explain why a bit. Sometimes we pass down the kind of decl `cannot be applied to a computer property` and it gives just enough context for a google search to return useful information.

Reply:
There are facilities to print a textual version of the decl kind in the diagnostics already, I can't find it at the moment.

Reply:
Good idea, I'll dig around to see if I can find that.

Reply:
Looked into this but it seems like improving the diagnostic would improve it for every attribute which would affect a ton of tests. I'll file a follow up radar to investigate doing this systemically.